data/mods/gen6/moves.ts

```diff
@@ -31,10 +39,9 @@ export const Moves: {[k: string]: ModdedMoveData} = {
 		condition: {
 			duration: 3,
 			onStart(target) {
-				const noEncore = ['encore', 'mimic', 'mirrormove', 'sketch', 'struggle', 'transform'];
```

Comment:
(I think what happened here is that the `noEncore` list diverged for Gen 7, and then the code was updated for the mechanics in later generations, but since those mechanics won't exist in Gen 6, the Gen 9 code would actually work with the move flag-based approach. I notice you already did the assist cases, although those are easier because it's not a condition.)